src/FakeItEasy/Creation/DummyValueResolver.cs

```diff
@@ -102,10 +102,16 @@ public ResolveByCreatingFakeStrategy(IFakeObjectCreator fakeCreator)
             private IFakeObjectCreator FakeCreator { get; }
 
             public override CreationResult TryCreateDummyValue(
-                    Type typeOfDummy,
-                    IDummyValueResolver resolver,
-                    LoopDetectingResolutionContext resolutionContext) =>
-                this.FakeCreator.CreateFake(typeOfDummy, new ProxyOptions(), resolver, resolutionContext);
+                Type typeOfDummy,
+                IDummyValueResolver resolver,
+                LoopDetectingResolutionContext resolutionContext)
+            {
+                return this.FakeCreator.CreateFakeWithoutCheckingIfWeAlreadyAre(
+                    typeOfDummy,
+                    new ProxyOptions(),
+                    resolver,
+                    resolutionContext);

```

Comment:
I think I might be missing the whole point of this method. Why is it safe in this case to not check if we're already resolving `typeOfDummy`?

Reply:
Ah, I see. Thanks for the explanation.